BgsService/BgsFactionData.cs

```diff
@@ -67,29 +66,16 @@ public List<Faction> GetFactions(string endpoint, List<KeyValuePair<string, obje
 
                 if (responses?.Count > 0)
                 {
-                    List<Faction> factions = ParseFactionsAsync(responses);
+                    List<Faction> factions = ParseFactionsParallel(responses);
                     return factions.OrderBy(x => x.name).ToList();
                 }
             }
             return null;
         }
 
-        private List<Faction> ParseFactionsAsync(List<object> responses)
+        private List<Faction> ParseFactionsParallel(List<object> responses)
         {
-            List<Task<Faction>> factionTasks = new List<Task<Faction>>();
-            foreach (object response in responses)
-            {
-                factionTasks.Add(Task.Run(() => ParseFaction(response)));
-            }
-            Task.WhenAll(factionTasks.ToArray());
-
-            List<Faction> factions = new List<Faction>();
-            foreach (Task<Faction> task in factionTasks)
-            {
-                Faction faction = task.Result;
-                if (faction != null) { factions.Add(faction); };
-            }
-
+            List<Faction> factions = responses.AsParallel().Select(ParseFaction).ToList();
```

Comment:
If an exception is thrown, the list may include a null value. 

Consider revising to 
```cs
List<Faction> factions = responses.AsParallel().Select(ParseFaction).OfType<Faction>().ToList();
```
to filter out null values?

Reply:
Good call, I'll add some tests while I'm at it. I think I prefer `Where(x => x != null)` as it makes the intention clearer.

Reply:
Upon further inspection and testing, the null value is handled higher up the call stack, so this is OK.

Reply:
If the faction is null, won't it fail at [line 70](https://github.com/EDCD/EDDI/pull/1706/files#diff-9f33a68b2e345eeaa36fb16a82b17dbbR70) above?

Reply:
`OrderBy()` can handle null elements, so it's OK.

Reply:
Proved by test. Ok. :-)